Makefile

```diff
@@ -0,0 +1,29 @@
+# Makefile
+THIS_FILE := $(lastword $(MAKEFILE_LIST))
+include common.mk
+SHELL := /bin/bash
+
+RUNTEST=python -m unittest -v -b
+
+ALLMODULES=$(patsubst %.py, %, $(wildcard test_*.py))
+
+all:
+	@echo "running tests..."
+	${RUNTEST} ${ALLMODULES}
+
+% : test_%.py
+	${RUNTEST} test_$@
+
+venv: venv/bin/activate
+
+venv/bin/activate: requirements.txt
+	test -d .venvbag || virtualenv --python=python3.6 .venvbag
+	.venvbag/bin/pip install -Ur requirements.txt
+	touch .venvbag/bin/activate
```

Comment:
What is the purpose of this?

Reply:
deployment of the lambda depends on python packages which are not standard. usually this is done by creating a virtual environment and installing those packages (which are listed in `requirements.txt`). This construct creates such a virtual environment, and installs requirements and deploys the lambda inside of it. This way user action is minimized.

Reply:
Sorry, I was not clear -- what is the purpose of the `touch` line? I understand the virtualenv and requirements part.